### PR TITLE
change `env` prefixes to XGENOME_

### PR DESCRIPTION
**Pull request by nnutter**
In order to break ties with the legacy config system I am changing the prefix for environment variables from `GENOME_` to `XGENOME_`.  This may mean that some scripts start behaving differently due to configuration not taking effect but it will allow us to immediately avoid the confusion of whether a value was set by our default or by the user and will allow us to change defaults without having to tell users to start new shells, etc.

### CODE REVIEW

**Comment by nnutter** [comment]
I've prepared `cle`, `cle_research`, `apipe-tester`, and `apipe-builder`, see,

```
head -n 100 ~{cle,cle_research,apipe-tester,apipe-builder}/.bashrc.d/genome.sh
```

and `imp-apipe` (and LIMS equivalent), see,

```
ssh imp-apipe.gsc.wustl.edu grep GENOME_ /etc/init.d/genome_view
```

**Comment by nnutter** [comment]
I'll add a changelog announce the environment variable changes when I merge.

File: genome-webserver/solr/scripts/gcsearch_load_from_db.sh [hunk 1]
```diff
@@ -10,21 +10,21 @@ set -o nounset   # fail if an env var is used but unset
 
 
 
-bsub -u jlolofie@genome.wustl.edu -q apipe GENOME_DEV_MODE=1 perl /gsc/scripts/bin/gcsearch_load_from_db --add user,individual,sample,population_group,taxon --lock 2
-bsub -u jlolofie@genome.wustl.edu -q apipe GENOME_DEV_MODE=1 perl /gsc/scripts/bin/gcsearch_load_from_db --add processing_profile,model_group,library,work_order --lock 3
+bsub -u jlolofie@genome.wustl.edu -q apipe XGENOME_DEV_MODE=1 perl /gsc/scripts/bin/gcsearch_load_from_db --add user,individual,sample,population_group,taxon --lock 2
+bsub -u jlolofie@genome.wustl.edu -q apipe XGENOME_DEV_MODE=1 perl /gsc/scripts/bin/gcsearch_load_from_db --add processing_profile,model_group,library,work_order --lock 3
```
Comment (tmooney) [inline genome-webserver/solr/scripts/gcsearch_load_from_db.sh]: I bet `jlolofie` is not reading those e-mails any time soon!

**Comment by tmooney** [comment]
+1 :mailbox: 

I still see `$ENV{GENOME_BUILD_ID}` (and `$ENV{GENOMEX_GMT_GENE_PREDICTOR_SNAP_LSF}`), but those aren't part of config.

**Comment by nnutter** [comment]
I didn't know about `GENOMEX_GMT_GENE_PREDICTOR_SNAP_LSF`.  I'll make a separate PR to convert that to `Genome::Config`.

**Comment by nnutter** [comment]
I'll need to merge/rebase once genome/genome#762 is merged.

**Comment by nnutter** [comment]
Merged in `master` and moved `gmt_gene_predictor_snap_lsf.yaml`.

**Comment by nnutter** [comment]
CI, test this.  `master` has fixed `Genome/Model/Tools/DetectVariants2/GatkGermlineSnv.t`.

**Comment by nnutter** [comment]
And I just accidentally merged without the changelog...

CHANGELOG: Almost all environment variables used to configure Genome have been changed.  If you have any scripts, etc. outside of the Genome Git repo then you'll need to update them to use the new Config API.  In addition to the documentation in the source code I've also started a [Confluence page](https://confluence.gsc.wustl.edu/display/APIPE/Configuration+Recipes).

**Comment by nnutter** [comment]
I pushed an empty commit straight to `master`:

```
commit 3af6367eae2fd425c1bfead89dc419cdaaca3600 (HEAD, origin/master, origin/HEAD, config)
Author: Nathaniel Nutter <nnutter@genome.wustl.edu>
Date:   Fri May 22 11:25:34 2015 -0500

    empty commit to add changelog
    
    CHANGELOG: Almost all environment variables used to configure Genome
    have been changed.  If you have any scripts, etc. outside of the Genome
    Git repo then you'll need to update them to use the new Config API.  In
    addition to the documentation in the source code I've also started a
    [Confluence page](https://confluence.gsc.wustl.edu/display/APIPE/Configuration+Recipes).
```

**Pull request merged**